### PR TITLE
Notify solvers if driver fails for some reason

### LINKED ISSUE #2140: bug: We don't `/notify` solvers on some errors
### Problem
It looks like there are some errors for which we don't correctly call the `/notify` endpoint to give the solver debugging information.

### Impact
External solver maintainers have a hard time debugging issues with their solutions.

### To reproduce
Check these [logs](https://production-6de61f.kb.eu-central-1.aws.cloud.es.io/app/discover#/?_g=(filters:!(),refreshInterval:(pause:!t,value:0),time:(from:now-7d%2Fd,to:now))&_a=(columns:!(log),filters:!(('$state':(store:appState),meta:(alias:!n,disabled:!f,index:c0e240e0-d9b3-11ed-b0e6-e361adffce0b,key:kubernetes.labels.app_kubernetes_io%2Finstance,negate:!t,params:(query:ingress-nginx),type:phrase),query:(match_phrase:(kubernetes.labels.app_kubernetes_io%2Finstance:ingress-nginx))),('$state':(store:appState),meta:(alias:!n,disabled:!f,index:c0e240e0-d9b3-11ed-b0e6-e361adffce0b,key:kubernetes.labels.appClass,negate:!t,params:(query:mainnet-api-prod),type:phrase),query:(match_phrase:(kubernetes.labels.appClass:mainnet-api-prod)))),grid:(columns:('@timestamp':(width:164))),index:ea511870-d9b3-11ed-a9d0-a17451f01cc1,interval:auto,query:(language:kuery,query:'log:%20%22notify%22'),sort:!(!('@timestamp',desc)))) to see that we don't have a single notification for:
* simulation errors
* negative scores
* encoding errors

### Expected behaviour
A solver should receive a `/notify` request for every solution it provided. Either a reason why it was discarded or the rank it achieved in the competition.

### PR DESCRIPTION
**Pull request by fleupold**
# Description
We don't always notify solvers if their solutions fails in certain cases (e.g. if we cannot generate an access list for their solution). However for debugging, a solver should receive a /notify request for every solution it provided that doesn't end up ranking.

# Changes
- [ ] Make some types of simulation errors that are related to the underlying transaction failing simulation classifying as such
- [ ] For remaining errors, still notify solvers but classify that this was a "driver" error (ie not their fault)

## How to test
CI

## Related Issues
Fixes #2140

### CODE REVIEW

File: crates/driver/src/infra/simulator/mod.rs [hunk 1]
```diff
@@ -189,14 +191,14 @@ where
                     None
                 }
             }
-            SimulatorError::Blockchain(blockchain::Error::Gas(_)) => None,
-            SimulatorError::Blockchain(blockchain::Error::Response(_)) => None,
+            SimulatorError::Blockchain(blockchain::Error::Gas(_)) => Some(tx),
+            SimulatorError::Blockchain(blockchain::Error::Response(_)) => Some(tx),
```
Comment (sunce86) [inline crates/driver/src/infra/simulator/mod.rs:195]: Why these three are now classified as revert errors?

File: crates/driver/src/infra/simulator/mod.rs [hunk 2]
```diff
@@ -180,7 +182,7 @@ where
             {
                 Some(tx)
             }
```
Comment (MartinquaXD) [inline crates/driver/src/infra/simulator/mod.rs:184]: This special case is not identical to the general case right below.

File: crates/driver/src/infra/simulator/mod.rs (hunk 1 above)
Comment (MartinquaXD) [reply crates/driver/src/infra/simulator/mod.rs:195]: Yeah, these seem more like they should be `Other` errors. Meaning it's entirely possible that the simulated transaction is fine but we just encountered an internal error.

Comment (fleupold) [reply crates/driver/src/infra/simulator/mod.rs:195]: You are right about Gas (I thought it was gas amount estimation related, but it's purely gas price related which the transaction has no influence on). 
However for `blockchain::Error::Response` it only happens when we fail to estimate the `accessList` which happens if the transaction doesn't simulate properly.

Will rename these variants to make that more clear.

File: crates/driver/src/infra/simulator/mod.rs (hunk 2 above)
Comment (fleupold) [reply crates/driver/src/infra/simulator/mod.rs]: right, reverted

**Review by MartinquaXD — APPROVED** [review]

**Review by sunce86 — APPROVED** [review]

**Pull request merged**